relentless/potential/potential.py

```diff
@@ -40,29 +41,33 @@ class CoefficientMatrix(PairMatrix):
 
     Set coefficient matrix values::
 
-        m['A','A']['energy'] = 2.0
-        m['A,'B']['mass'] = 1.5
-        m['A','B']['energy'] = 3.0
+        m['A','A']['energy'] = 2.0               #access param directly
+        m['A','B'] = {'energy':0.0, 'mass':1.5}  #reset default in full
+        m['A','A'] = {'mass':0.5}                #reset default partially
+        for p in self.params:                    #iteratively acesss params
+            m['B','B'][p] = 0.1
 
     Evaluate (retrieve) pair parameters::
 
         >>> print(m.evaluate(('A','A')))
-        {'energy':2.0, 'mass':2.0}
+        {'energy':2.0, 'mass':0.5}
         >>> print(m.evaluate(('A','B')))
-        {'energy':3.0, 'mass':1.5}
+        {'energy':0.0, 'mass':1.5}
         >>> print(m.evaluate(('B','B')))
-        {'energy':1.0, 'mass':2.0}
+        {'energy':0.1, 'mass':0.1}
 
     """
     def __init__(self, types, params, default={}):
         super().__init__(types)
         if not all(isinstance(p, str) for p in params):
             raise TypeError('All parameters must be strings')
         self.params = tuple(params)
+
+        d = FixedKeyDict(keys=self.params)
+        for i in d:
+            d[i] = default[i] if i in default else None
         for key in self:
-            for p in self.params:
-                v = default[p] if p in default else None
-                self[key][p] = v
+            self[key] = d
```

Comment:
I think you need to create `FixedKeyDict` here inside the loop, rather than once above. As it reads now, I am worried that python might assign the _reference_ to the same object to `self[key]`, which means a change to one will change them all. Your unit tests look like they check for this, so maybe I am mistaken Alternatively, we could implement a `copy` method for `FixedKeyDict`.